Automattic/jetpack
# GDPR: Module/Feature Privacy Links

Ref: p1HpG7-50l-p2

This PR adds "Privacy Information" links into each Jetpack module/feature.

![1](https://user-images.githubusercontent.com/1563559/38153986-08347d34-341c-11e8-85c2-803947940fbe.png)

![2](https://user-images.githubusercontent.com/1563559/38153987-085c129a-341c-11e8-9761-96444dd9cb5a.png)

![3](https://user-images.githubusercontent.com/1563559/38153988-08824500-341c-11e8-9b67-a5c40a8f1197.png)

### Are the "Privacy Information" links specific/explicit enough?

Please express your thoughts about this.

### Are Privacy Information links working yet?

In most cases, the existing `learn_more_button` link is simply appended with a `#privacy` hash. The `#privacy` section in each document will come soon, at which time the links will work automatically. 

### How to Test this PR

- Merge this branch and: `npm run build`
- Enable all Jetpack functionality on a Professional plan from the debug page in Jetpack:
  `/wp-admin/admin.php?page=jetpack_modules`
- Go to the Jetpack Dashboard page and check all of the icons in each module/feature to ensure they are working properly. In the case of "Privacy Information" links that aren't supposed to be working yet, please just review the URL itself and report any problems.
- Repeat for every module/feature in the Jetpack Settings page too.

### Overview of Changes

- Adds a new `SupportInfo` component. See [README](https://github.com/Automattic/jetpack/tree/add/gdpr-feature-privacy-links/_inc/client/components/support-info) in the [`_inc/client/components/support-info`](https://github.com/Automattic/jetpack/tree/add/gdpr-feature-privacy-links/_inc/client/components/support-info) directory. The new component is used to display a support info icon. When clicked, it displays a popover with a description, a learn more link, and a privacy info link.

  This new component is a way of making what was `.jp-module-settings__learn-more` into a reusable component, and therefore easier to adjust globally.

- Removes `.jp-module-settings__learn-more` from the `SettingsGroup` component in favor of the new `SupportInfo` component.

- Updates CSS adjustments across a few components to change references from `.jp-module-settings__learn-more` into `.jp-support-info`.

- Adds support info icons to all `DashItem` instances on the Dashboard page, along with every module/feature in the Settings page.
